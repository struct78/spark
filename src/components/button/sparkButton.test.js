import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SparkButton from './sparkButton';

describe("Button", () => {
  const wrapper = shallow(<SparkButton onPress={()=> { return true; }}>This is a button</SparkButton>);

  it ("renders correctly", () => {
    expect(renderer.create(wrapper).toJSON()).toMatchSnapshot();
  });

  it ("creates text child correctly", () => {
    expect(wrapper.text()).toEqual('This is a button');
  });

  it ("onPress fires correctly", () => {
    expect(wrapper.props().onPress()).toBe(true);
  });
});
