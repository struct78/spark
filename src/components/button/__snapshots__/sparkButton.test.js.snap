// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "marginRight": 2,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 20,
        },
        null,
      ]
    }
  >
    This is a button
  </Text>
</View>
`;
